config/check_std_isnan.cxx

```diff
@@ -0,0 +1,8 @@
+#include <cmath>
+int main()
+{
+    // Some old versions of libc++ do not provide isnan() for integers,
+    //  in which case this will fail to compile.
+    std::isnan(int(0));
```

Comment:
This is strange: why doesn't the compiler perform an implicit conversion to `double` here? In fact, cppreference (http://en.cppreference.com/w/cpp/numeric/math/isnan) explicitly says that this would happen. Maybe the actual problem is something else?